### PR TITLE
Add `enableOrderFormOptimization` setting to the app

### PR DESCRIPTION
**Pull request by victorhmp**
#### What is the purpose of this pull request?

Adding a new `enableOrderFormOptimization` setting to the app, which enables users to control whether or not the "legacy" `OrderFormProvider` (the one that does not use `checkout-graphql`) is rendered by the `StoreWrapper` component.

#### What problem is this solving?

This should enable stores to be faster by not having two different orderForms being loaded.

#### How should this be manually tested?

1. Go to [Workspace with enableOrderFormOptimization = `true`](https://victorhmp--storecomponents.myvtex.com/), where this setting is set to `true`.
2. Open Apollo DevTools and check the queries being performed to render the page. You should see only one mention of `orderForm`, which is a query performed by the *new* OrderFormProvider to `checkout-graphql`.
3. Go to [Workspace with enableOrderFormOptimization = `false`](https://victormiranda--storecomponents.myvtex.com/), where this setting was not set (using the default `false` value).
4. Repeat step 2., but this time you should be able to spot **two** `orderForm` queries being performed. Each one of them used by a different OrderFormProvider.

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

#### Notes

This is related to: https://github.com/vtex-apps/admin-pages/pull/329 and https://github.com/vtex-apps/io-documentation/pull/524

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -99,6 +99,11 @@
         "title": "Search Term Path",
         "type": "string",
         "description": "Indicate the search path of your store"
+      },
+      "legacyOrderForm": {
+        "title": "Use legacy orderForm provider",
+        "type": "boolean",
+        "description": "Whether or not the legacy OrderFormProvider component should be rendered by your store. Setting this to true could hurt performance"
```
Comment (victorhmp) [inline manifest.json]: Not fully sold on this description

Comment (kaisermann) [reply manifest.json]: `could lead to a less smooth browsing experience`?

File: manifest.json [hunk 2]
```diff
@@ -88,17 +88,19 @@
...
       },
       "searchTermPath": {
         "title": "Search Term Path",
         "type": "string",
         "description": "Indicate the search path of your store"
+      },
+      "legacyOrderForm": {
+        "title": "Use legacy orderForm provider",
+        "type": "boolean",
+        "description": "Whether or not the legacy OrderFormProvider component should be rendered by your store. Setting this to true could lead to a less smooth browsing experience."
```
Comment (klzns) [inline manifest.json]:
```suggestion
        "description": "Whether or not the legacy OrderFormProvider component should used in your store. <a href="https://vtex.io/docs/">Read more about it.</a>"
```

**Comment by victorhmp** [comment]
Working on documentation to support this new setting

File: react/StoreWrapper.js [hunk 3]
```diff
@@ -108,6 +109,17 @@ const StoreWrapper = ({ children }) => {
 
   const parsedFavicons = useFavicons(faviconLinks)
 
+  const childrenWithNewOrderForm = (
```
Comment (lbebber) [inline react/StoreWrapper.js]: coloca um comentario aqui e lá embaixo sobre o que significa esse negocio de new order form

Comment (lbebber) [reply react/StoreWrapper.js]: na vdd tb nao precisa dizer que é "with new order form", isso aqui é meramente o `content`, o que é extra é o fato de adicionar um order form legacy lá embaixo

**Pull request merged**